mongo/integration/crud_prose_test.go

```diff
@@ -327,88 +324,6 @@ func TestHintErrors(t *testing.T) {
 	})
 }
 
-type testValueMarshaler struct {
-	val []bson.D
-}
-
-func (tvm testValueMarshaler) MarshalBSONValue() (bsontype.Type, []byte, error) {
-	return bson.MarshalValue(tvm.val)
-}
-
-func TestAggregatePrimaryPreferredReadPreference(t *testing.T) {
```

Comment:
Looks like this table test was added with GODRIVER-1782 to test that `$out` is properly detected and `$readPreference` is not included in the corresponding command. The new spec tests in this PR test that this behavior is still true on wire version < 13, so I believe this test is redundant (it is also not a CRUD prose test despite its location in `crud_prose_test.go`).